lib/CMakeLists.txt

```diff
@@ -51,12 +51,13 @@ endif()
 
 # The rest of the build
 
-include_directories(${CMAKE_CURRENT_BINARY_DIR}/../include)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/..)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../include)
-include_directories(${CMAKE_CURRENT_BINARY_DIR}/..)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR})
-include_directories(${CMAKE_CURRENT_BINARY_DIR})
+include_directories(
+  ${CURL_BINARY_DIR}/include/curl # for generated curlbuild.h from curl.h
```

Comment:
I think this was done for consistency with some other locations where this include directory is changed.

Using `project(CURL)` will set the variables CURL_SOURCE_DIR and CURL_BINARY_DIR, see https://cmake.org/cmake/help/v3.0/command/project.html. Maybe this is used in case you have multiple projects in a single source tree (although I have never used that feature before).